template_handler/cdk.py

```diff
@@ -0,0 +1,218 @@
+        """
+
+        # Check Command Prerequisites
+        cmd_prerequisites = [
+            'node',
+            'npx'
+        ]
+        for cmd_prerequisite in cmd_prerequisites:
+            if not self._cmd_exists(cmd_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Command prerequisite '{cmd_prerequisite}' not found")
+
+        # Check Node Package Prerequisites
+        node_prerequisites = [
+            'cdk-assets'
+        ]
+        for node_prerequisite in node_prerequisites:
+            if not self._node_package_exists(node_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Node Package prerequisite '{node_prerequisite}' not found")
+
+    def handle(self) -> str:
+        """
+        Main Sceptre CDK Handler function
+
+        Returns
+        -------
+        str|bytes
+            CDK synthesised CloudFormation template
+        """
+
+        self.logger = self.PrefixLoggerAdapter(self.logger, {'prefix': self.name})
+
+        self._check_prerequisites()
+
+        # Import CDK Python template module
+        template_path = str(pathlib.Path('templates', self.cdk_template_path))
+        self.logger.debug(f'Importing CDK Python template module {template_path}')
+        template_module_name = pathlib.Path(template_path).stem
+        loader = importlib.machinery.SourceFileLoader(template_module_name, template_path)
+        spec = importlib.util.spec_from_loader(template_module_name, loader)
+        template_module = importlib.util.module_from_spec(spec)
+        loader.exec_module(template_module)
+
+        # CDK Synthesize App
+        self.logger.debug(f'CDK synthesing CdkStack Class')
+        self.logger.debug(f'CDK Context: {self.cdk_context}')
+        app = aws_cdk.App(context=self.cdk_context)
+        stack_name = 'CDKStack'
+        template_module.CdkStack(app, stack_name, self.sceptre_user_data)
+        app_synth = app.synth()
+
+        # Publish CDK Assets
+        asset_artifacts = None
+
+        for artifacts in app_synth.artifacts:
+            if isinstance(artifacts, aws_cdk.cx_api.AssetManifestArtifact):
+                asset_artifacts = artifacts
+                break
+        if asset_artifacts is None:
+            raise exceptions.SceptreException(f'{self.name} - CDK Asset manifest artifact not found')
+        environment_variables = self._get_envs()
+        self.logger.info(f'Publishing CDK assets')
+        self.logger.debug(f'Assets manifest file: {asset_artifacts.file}')
+        cdk_assets_result = subprocess.run(
+            f'npx cdk-assets publish --path {asset_artifacts.file}',
+            env=environment_variables,
+            shell=True,
+            capture_output=True)
+        self.logger.info(f'{cdk_assets_result.stderr.decode()}')
+        if cdk_assets_result.returncode != 0:
+            raise exceptions.SceptreException(f'{self.name} - Error publishing CDK assets')
```

Comment:
This could be simplified by doing this
```suggestion
        cdk_assets_result = subprocess.run(
            f'npx cdk-assets publish --path {asset_artifacts.file}',
            env=environment_variables,
            shell=True,
            capture_output=True,
            stdout=sys.stderr,  # By default stderr will already get output to the parent process's stderr. 
            check=True  # This will raise an error if the return code isn't zero
        )
```